spec/components/search_spec.rb

```diff
@@ -236,6 +236,54 @@
 
     end
 
+    context 'personal-direct flag' do
+      let(:current) { Fabricate(:user, admin: true, username: "current_user") }
+      let(:participant) { Fabricate(:user, username: "participant_1") }
+      let(:participant_2) { Fabricate(:user, username: "participant_2") }
+
+      let(:group) do
+        group = Fabricate(:group, has_messages: true)
+        group.add(current)
+        group.add(participant)
+        group
+      end
+
+      def create_pm(users:, group: nil)
+        pm = Fabricate(:private_message_post, user: users.first).topic
+        pm.topic_allowed_users.where.not(user_id: users.first.id).destroy_all
```

Comment:
It's better to construct a new object that has the data you want rather than using an existing one and deleting the stuff you don't.